### ISSUE TITLE
Melee units as defence

### ISSUE THREAD

**Issue by ByBurton** [Sh0wdown/UnofficialCrusaderPatch#318]
So I have been tinkering around a lot with the AIV Editor and my AICE.
In the AIV Editor you can place points for melee units. But they don't behave the same.
What I have seen so far:
1. Spearmen and Macemen patrol between the different placed points. Attack anything in range.
2. Horse archers (not being melee, but not being stationary) also do, but in groups.
3. Slaves take up position at the points, same as knights. They attack anything in range.
4. Other units, like both types of swordsmen or pikemen, don't do either of those three.

So my question is: Can this forth point either be changed to make all of those patrol, or at least take up position? I would love seeing Richard have pikemen on his walls, or Emir swordsmen on his walls.

**Comment by Heroesflorian** [comment]
> Slaves take up position at the points, same as knights. They attack anything in range.

Interesting... one curious thing I have observed in the past (pre-UCP) was that many (all?) AI characters would, if you add slave spots to their AIVs, put their regular defensive ranged troops on the slave spots instead of their regular spots. 
E.g. Saladin would put some fire throwers AND archers on his slave spots, but then not man his archer / fire thrower spots fully.

**Comment by ByBurton** [comment]
> Slaves take up position at the points, same as knights. They attack anything in range.
> 
> Interesting... one curious thing I have observed in the past (pre-UCP) was that many (all?) AI characters would, if you add slave spots to their AIVs, put their regular defensive ranged troops on the slave spots instead of their regular spots.
> E.g. Saladin would put some fire throwers AND archers on his slave spots, but then not man his archer / fire thrower spots fully.

I can not confirm that, and I tested it quite a lot.

**Comment by ByBurton** [comment]
Update: Assassins are also sent to the spots, same as Slaves.
Leaves only swordsmen and pikemen....
I'd really love to see this feature.

**Comment by Panbutt** [comment]
> > Slaves take up position at the points, same as knights. They attack anything in range.
> 
> Interesting... one curious thing I have observed in the past (pre-UCP) was that many (all?) AI characters would, if you add slave spots to their AIVs, put their regular defensive ranged troops on the slave spots instead of their regular spots.
> E.g. Saladin would put some fire throwers AND archers on his slave spots, but then not man his archer / fire thrower spots fully.
Yeah I thought this was always the case no matter the AI. I've never seen slaves man spots themselves, only units meant for other spots.

**Comment by ByBurton** [comment]
> Yeah I thought this was always the case no matter the AI. I've never seen slaves man spots themselves, only units meant for other spots.

Actually no. I made an AI which uses defensive slaves and defensive slingers. The slaves take up position on their spots, the slingers on theirs.
Not any other units are sent to the slave spots.

**Comment by ByBurton** [comment]
@Sh0wdown Could you let me know if you could somehow just copy the behaviour of spearmen, macemen and horse archers to all other melee units? 
Or copy the behaviour of assassins / slaves / knights / archers / etc.  to pikemen, swordsmen, arabian swordsmen?

**Comment by KingCaliph** [comment]
Where can I download the AICE Editor???? Also, when you refer to AIV Editor do you mean the AI Village download? or some other editor I am missing out on???

**Comment by ByBurton** [comment]
@KingCaliph the aice editor is dead. I stopped developing it.

**Comment by J-T-de** [comment]
Since I mentioned this in another issue and saw this question @KingCaliph:
This is probably something you are looking for https://github.com/CIO61/ucp_re_gui/releases.
